keras_nlp/samplers/sampler.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base sampler class."""
+
+import tensorflow as tf
+
+
+class Sampler:
+    """Base sampler class.
```

Comment:
I might document this class explaining how it could be subclasses, similar to `Tokenizer`. Even though we are not exposing it right now, we could in the future, and explaining it's intended usage would be helpful.

Reply:
Sounds good! But I would just cut this docstring down then. If we aren't ready to describe subclassing, we probably should not put the base class in the release (which seems totally fine).

So we should focus our actual runnable example documentation of the symbols themselves.